src/common/snippets/src/lowered/pass/insert_perf_count.cpp

```diff
@@ -5,56 +5,52 @@
 
 #include "snippets/lowered/pass/insert_perf_count.hpp"
 #include "snippets/lowered/linear_ir.hpp"
-#include "snippets/snippets_isa.hpp"
 #include "snippets/itt.hpp"
 
 namespace ov {
 namespace snippets {
 namespace lowered {
 namespace pass {
 
+InsertPerfCount::InsertPerfCount(std::map<std::string, std::string> boundary_op_names)
+    : Pass(), m_boundary_op_names(std::move(boundary_op_names)) {
+}
+
 bool InsertPerfCount::run(LinearIR& linear_ir) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InsertPerfCount")
     if (linear_ir.empty())
         return false;
+    if (m_boundary_op_names.empty()) {
+        const auto& first_op_name = linear_ir.begin()->get()->get_node()->get_friendly_name();
+        const auto& last_op_name = linear_ir.rbegin()->get()->get_node()->get_friendly_name();
+        m_boundary_op_names.insert({first_op_name, last_op_name});
+    }
 
-    auto is_parameter = [](const std::shared_ptr<ov::Node>& node) {
-        return ov::is_type<ov::op::v0::Parameter>(node);
-    };
-    auto is_result = [](const std::shared_ptr<ov::Node>& node) {
-        return ov::is_type<ov::op::v0::Result>(node);
-    };
-
-    // mark perf_count_begin and perf_count_end position
-    auto perf_count_begin_pos = linear_ir.cbegin();
-    auto perf_count_end_pos = perf_count_begin_pos;
-    bool first_result_marked = false;
+    size_t seq_number = 0;
     for (auto expr_it = linear_ir.cbegin(); expr_it != linear_ir.cend(); expr_it++) {
-        const auto expr = *expr_it;
-        const auto& node = expr->get_node();
-        if (is_parameter(node))
-            perf_count_begin_pos = expr_it;
-
-        if (is_result(node) && !first_result_marked) {
-            perf_count_end_pos = expr_it;
-            first_result_marked = true;
+        const auto& op_name = expr_it->get()->get_node()->get_friendly_name();
+        const auto& found = m_boundary_op_names.find(op_name);
+        if (found != m_boundary_op_names.end()) {
+            const auto perf_count_begin_pos = expr_it;
+            auto perf_count_end_pos = expr_it;
+            while (perf_count_end_pos->get()->get_node()->get_friendly_name() != found->second &&
+                   perf_count_end_pos != linear_ir.cend()) {
+                perf_count_end_pos++;
+            }
+            OPENVINO_ASSERT(perf_count_end_pos != linear_ir.cend(), "Failed to find requested op name to insert PerfCountEnd");
+            // insert perf_count_begin after last parameter
```

Comment:
Minor: it seems like the comment is not actual anymore.

Reply:
Ok, removed